Restores the original default format of search slow log

imotov opened the pull request:
In 5.0, the search slow log switched to the multi-line format with no option to get back to the origin single-line format that was used prior to 5.0 by default. This commit changes the default back to the single-line format and restores an option of switching to the multi-line format if needed.

Closes #21711 

Not sure if this should be treated as bug or as an enhancement and therefore if we should document it in breaking changes or not (if this is a bug, we are fixing it, if this is an enhancement, we are breaking it) and if this is a bug, should we push it to 5.1.0, 5.1.1 or 5.2.0 is good enough.

clintongormley commented:
I think this would qualify as a regression, and should go back to 5.0 and 5.1

clintongormley commented:
Discussed in FixItFriday - let's remove the setting, just make it single line all the time.

imotov commented:
I removed the `reformat` setting from slow search log and added a test to make sure we always log a single line. I kept and documented the `reformat` setting for the index slow log though. I think it makes more sense there and can be actually quite useful. In the search log the original source is gone by the time we reach search and it makes little sense to pretty print it. However, in the indexing case the original format of the record that we index is preserved all the way through, so it might be beneficial to have an ability to check the original format that was sent to elasticsearch.

imotov commented:
@javanna would you mind reviewing it?

javanna commented on core/src/main/java/org/elasticsearch/index/SearchSlowLog.java:
```diff
@@ -73,20 +73,17 @@
         Setting.timeSetting(INDEX_SEARCH_SLOWLOG_PREFIX + ".threshold.fetch.trace", TimeValue.timeValueNanos(-1),
             TimeValue.timeValueMillis(-1), Property.Dynamic, Property.IndexScope);
-    public static final Setting<Boolean> INDEX_SEARCH_SLOWLOG_REFORMAT =
-        Setting.boolSetting(INDEX_SEARCH_SLOWLOG_PREFIX + ".reformat", true, Property.Dynamic, Property.IndexScope);
```
let me guess, we still had the setting but it had no effect? We were reading and passing it all the way to the log printer but we didn't check the flag value when needed? That is why also removing this setting should not be considered a breaking change, as the setting didn't have any effect?

imotov replied:
The setting had an effect before 5.x. In 2.x and below we had access to the original text of the query on the shard level and therefore we could choose to write it into the slow log in the original format (possibly with new lines) or reformat it into a single line. In 5.x the query comes to us in a parsed way, so the original query is not longer available hence it makes little sense to keep this setting. In other words, we broke it in 5.0.0, now we return back original default behavior while removing the setting...

javanna commented on core/src/main/java/org/elasticsearch/index/SearchSlowLog.java:
```diff
@@ -73,20 +73,17 @@
         new Setting<>(INDEX_SEARCH_SLOWLOG_PREFIX + ".level", SlowLogLevel.TRACE.name(), SlowLogLevel::parse, Property.Dynamic,
             Property.IndexScope);
 
+    private static final ToXContent.Params FORMAT_PARAMS = new ToXContent.MapParams(Collections.singletonMap("pretty", "false"));
```
isn't pretty false already the default?

javanna replied:
good point, I think I went for pretty as default there because it's called by `toString`, and that made more sense in most of the cases, at least for queries and java api builders. Those methods are there solely for logging and debugging at this point. It does make sense to pass in false then in this specific case.

javanna commented on core/src/test/java/org/elasticsearch/index/SearchSlowLogTests.java:
```diff
@@ -48,120 +50,93 @@ protected SearchContext createSearchContext(IndexService indexService) {
         assertThat(p.toString(), startsWith("[foo][0]"));
-    }
-
-    public void testReformatSetting() {
```
funny, we were testing the setting and the test succeeded although the setting was ignored? :/
